### PR TITLE
Updates of Service Events:  layout structure and link types

### PR DESCRIPTION
**Pull request by l-ohlen-ericsson**

### CODE REVIEW

File: eiffel-vocabulary/EiffelArtifactDeployedEvent.md [hunk 1]
```diff
@@ -13,7 +13,7 @@
    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
    See the License for the specific language governing permissions and
    limitations under the License.
---->
+---> 
 
 # EiffelArtifactDeployedEvent (ArtD)
 The EiffelArtifactDeployedEvent id to declare that an artifact has been deployed into an environment.
```
Comment (xvinosi-github) [inline eiffel-vocabulary/EiffelArtifactDeployedEvent.md]: is to declare or declares

File: eiffel-vocabulary/EiffelArtifactDeployedEvent.md [hunk 2]
```diff
@@ -25,25 +25,149 @@ __Type:__ String
 __Required:__ No  
 __Description:__  A URI where further information can be obtained, if applicable.
 
-## Legal Link Types
+## Legal Link Types
```
Comment (xvinosi-github) [inline eiffel-vocabulary/EiffelArtifactDeployedEvent.md]: ## Links as per eiffel repo

File: eiffel-vocabulary/EiffelServiceAllocatedEvent.md [hunk 3]
```diff
@@ -14,7 +14,7 @@
    See the License for the specific language governing permissions and
    limitations under the License.
 --->
-
+ 
 # EiffelServiceAllocatedEvent  (ServA)
 The EiffelServiceAllocatedEvent is to declare an instance has been allocated to a User.( e.g global scheduler or resource booking tool).
```
Comment (xvinosi-github) [inline eiffel-vocabulary/EiffelServiceAllocatedEvent.md]: that infrastructure needed for a Service Instance has 
Ex: A global scheduler or Resource booking tool has allocated the the resources to user

File: eiffel-vocabulary/EiffelServiceAllocatedEvent.md [hunk 4]
```diff
@@ -34,25 +34,142 @@ __Type:__ String
 __Required:__ No  
 __Description:__ A URI where further information can be obtained, if applicable.
 
-## Legal Link Types
+## Legal Link Types
```
Comment (xvinosi-github) [inline eiffel-vocabulary/EiffelServiceAllocatedEvent.md]: Links

File: eiffel-vocabulary/EiffelServiceDeployedEvent.md [hunk 5]
```diff
@@ -14,7 +14,7 @@
    See the License for the specific language governing permissions and
    limitations under the License.
 --->
-
+
```
Comment (xvinosi-github) [inline eiffel-vocabulary/EiffelServiceDeployedEvent.md:17]: The EiffelServiceDeployedEvent is to declare that  a previously allocated service instance ((declared by [EiffelServiceAllocatedEvent]) has been configured and deployed with required Artifacts by a machine/human.
 It implies that the service instance is ready to be started.

We could discuss more on data types.

File: eiffel-vocabulary/EiffelServiceDeployedEvent.md [hunk 6]
```diff
@@ -45,30 +45,162 @@ __Type:__ String
...
 
-## Legal Link Types
-### ENVIRONMENT
-__Legal targets:__ [EiffelEnvironmentDefinedEvent](https://github.com/Ericsson/eiffel/blob/master/eiffel-vocabulary/EiffelEnvironmentDefinedEvent.md)  
-__Multiple allowed:__ No  
-__Description:__ Identifies the environment in which an event occurred, e.g. in which environment an artifact was built or an issue was verified.
+## Legal Link Types 
+
+### ARTIFACT_DEPLOYED
+__Required:__ No
```
Comment (xvinosi-github) [inline eiffel-vocabulary/EiffelServiceDeployedEvent.md]: May be we could explain more on why required is no, like if the service allocator could find the matching service with all infrastructure already deployed we need not deploy anything

File: eiffel-vocabulary/EiffelServiceDeployedEvent.md [hunk 7]
```diff
@@ -45,30 +45,162 @@ __Type:__ String
...
-__Legal targets:__ [EiffelActivityTriggeredEvent](https://github.com/Ericsson/eiffel/blob/master/eiffel-vocabulary/EiffelActivityTriggeredEvent.md), 
+__Required:__ Yes  
+__Legal targets:__ [EiffelServiceAllocated](./EiffelServiceAllocated.md)  
 __Multiple allowed:__ No  
-__Description:__ Identifies the activity of which the event constitutes a part. SHOULD not be used in conjunction with __CAUSE__, see above. Note that multiple layers may be modeled using __CONTEXT__, e.g. an activity being part of another activity.
+__Description:__ Identifies the service that artifacts was deployed at
 
-### ACTIVITY_EXECUTION
-__Legal targets:__ [EiffelActivityTriggeredEvent](https://github.com/Ericsson/eiffel/blob/master/eiffel-vocabulary/EiffelActivityTriggeredEvent.md)  
+### ENVIRONMENT
```
Comment (xvinosi-github) [inline eiffel-vocabulary/EiffelServiceDeployedEvent.md]: ENVIRONMENT?

File: eiffel-vocabulary/EiffelServiceDeployedEvent.md [hunk 8]
```diff
@@ -45,30 +45,162 @@ __Type:__ String
...
+
+### FLOW_CONTEXT
+__Required:__ No  
+__Legal targets:__ [EiffelFlowContextDefinedEvent](https://github.com/Ericsson/eiffel/blob/master/eiffel-vocabulary/EiffelFlowContextDefinedEvent.md)  
+__Multiple allowed:__ Yes  
+__Description:__ Identifies the flow context of the event: which is the continuous integration and delivery flow in which this occurred - e.g. which product, project, track or version this is applicable to.
+
+
+### SERVICE
+__Required:__ Yes
```
Comment (xvinosi-github) [inline eiffel-vocabulary/EiffelServiceDeployedEvent.md]: Required could be NO.

If we do not have any orchestrator tool and fixed hardware, then we will not have a service allocated event.

File: eiffel-vocabulary/EiffelServiceDeployedEvent.md [hunk 9]
```diff
@@ -45,30 +45,162 @@ __Type:__ String
...
+### FLOW_CONTEXT
+__Required:__ No  
+__Legal targets:__ [EiffelFlowContextDefinedEvent](https://github.com/Ericsson/eiffel/blob/master/eiffel-vocabulary/EiffelFlowContextDefinedEvent.md)  
+__Multiple allowed:__ Yes  
+__Description:__ Identifies the flow context of the event: which is the continuous integration and delivery flow in which this occurred - e.g. which product, project, track or version this is applicable to.
+
+
+### SERVICE
+__Required:__ Yes  
+__Legal targets:__ [EiffelServiceDeployedEvent](./EiffelServiceDeployedEvent.md)
```
Comment (xvinosi-github) [inline eiffel-vocabulary/EiffelServiceDeployedEvent.md]: to service allocated.
You are also true, we could also point to previous service deployed event also.
may be we need a new link type to point to allocation like ALLOCATION

File: eiffel-vocabulary/EiffelServiceDeployedEvent.md [hunk 10]
```diff
@@ -45,30 +45,162 @@ __Type:__ String
...
+__Legal targets:__ [EiffelFlowContextDefinedEvent](https://github.com/Ericsson/eiffel/blob/master/eiffel-vocabulary/EiffelFlowContextDefinedEvent.md)  
+__Multiple allowed:__ Yes  
+__Description:__ Identifies the flow context of the event: which is the continuous integration and delivery flow in which this occurred - e.g. which product, project, track or version this is applicable to.
+
+
+### SERVICE
+__Required:__ Yes  
+__Legal targets:__ [EiffelServiceDeployedEvent](./EiffelServiceDeployedEvent.md)  
+__Multiple allowed:__ No  
+__Description:__ Identifies the service that have been stopped
```
Comment (xvinosi-github) [inline eiffel-vocabulary/EiffelServiceDeployedEvent.md]: allocation and not stopped.

File: eiffel-vocabulary/EiffelServiceDiscontinuedEvent.md [hunk 11]
```diff
@@ -29,25 +29,148 @@ __Type:__ String
 __Required:__ No  
 __Description:__ A URI where further information can be obtained, if applicable.
 
-## Legal Link Types
+## Legal Link Types
```
Comment (xvinosi-github) [inline eiffel-vocabulary/EiffelServiceDiscontinuedEvent.md:32]: Links

File: eiffel-vocabulary/EiffelServiceReturnedEvent.md [hunk 12]
```diff
@@ -30,25 +30,150 @@ __Type:__ String
...
+__Multiple allowed:__ Yes  
+__Description:__ Identifies the flow context of the event: which is the continuous integration and delivery flow in which this occurred - e.g. which product, project, track or version this is applicable to.
+
+
+### SERVICE
+__Required:__ Yes  
+__Legal targets:__ [EiffelServiceDeployedEvent](./EiffelServiceDeployedEvent.md)  
 __Multiple allowed:__ No  
-__Description:__ Declares the activity execution the event relates to. In other words, [EiffelActivityTriggeredEvent](https://github.com/Ericsson/eiffel/blob/master/eiffel-vocabulary/EiffelActivityTriggeredEvent.md) acts as a handle for the activity execution. This differs from __CONTEXT__. In __ACTIVITY_EXECUTION__ the source carries information pertaining to the target (i.e. the activity started, finished or was canceled). In __CONTEXT__, on the other hand, the source constitutes a subset of the target.
+__Description:__ Identifies the service that have been stopped
```
Comment (xvinosi-github) [inline eiffel-vocabulary/EiffelServiceReturnedEvent.md]: that has been returned back to Orchestrator.

File: eiffel-vocabulary/EiffelServiceStartedEvent.md [hunk 13]
```diff
@@ -34,25 +34,151 @@ __Type:__ String
 __Required:__ No  
 __Description:__ A URI where further information can be obtained, if applicable. 
 
-## Legal Link Types
+## Legal Link Types
```
Comment (xvinosi-github) [inline eiffel-vocabulary/EiffelServiceStartedEvent.md]: Links

File: eiffel-vocabulary/EiffelServiceStartedEvent.md [hunk 14]
```diff
@@ -14,7 +14,7 @@
    See the License for the specific language governing permissions and
    limitations under the License.
 --->
-
+ 
 # EiffelServiceStartedEvent (ServSta)
 The EiffelServiceStartedEvent is to declare that a Service has been started and is running.
```
Comment (xvinosi-github) [inline eiffel-vocabulary/EiffelServiceStartedEvent.md]: previosuly deployed (declared by [servicedeployedevent])

File: eiffel-vocabulary/EiffelServiceStartedEvent.md [hunk 15]
```diff
@@ -34,25 +34,151 @@ __Type:__ String
...
+__Multiple allowed:__ Yes  
+__Description:__ Identifies the flow context of the event: which is the continuous integration and delivery flow in which this occurred - e.g. which product, project, track or version this is applicable to.
+
+
+### SERVICE
+__Required:__ Yes  
+__Legal targets:__ [EiffelServiceDeployedEvent](./EiffelServiceDeployedEvent.md)  
 __Multiple allowed:__ No  
-__Description:__ Declares the activity execution the event relates to. In other words, [EiffelActivityTriggeredEvent](https://github.com/Ericsson/eiffel/blob/master/eiffel-vocabulary/EiffelActivityTriggeredEvent.md) acts as a handle for the activity execution. This differs from __CONTEXT__. In __ACTIVITY_EXECUTION__ the source carries information pertaining to the target (i.e. the activity started, finished or was canceled). In __CONTEXT__, on the other hand, the source constitutes a subset of the target.
+__Description:__ Identifies the service that have been stopped
```
Comment (xvinosi-github) [inline eiffel-vocabulary/EiffelServiceStartedEvent.md]: started

**Review by xvinosi-github — CHANGES_REQUESTED** [review]

File: eiffel-vocabulary/EiffelServiceDeployedEvent.md (hunk 6 above)
Comment (l-ohlen-ericsson) [reply eiffel-vocabulary/EiffelServiceDeployedEvent.md]: Setting it to Yes, would simply add a strict constraint that usually is avoided as I see it.  But we can for sure discuss this

File: eiffel-vocabulary/EiffelServiceDeployedEvent.md (hunk 9 above)
Comment (l-ohlen-ericsson) [reply eiffel-vocabulary/EiffelServiceDeployedEvent.md]: Time is not on our side....

**Pull request merged**